threescale/src/utils.rs

```diff
@@ -75,3 +80,42 @@ pub fn add_hierarchy_to_metrics(hierarchy: &Hierarchy, metrics: &mut Metrics) {
         }
     }
 }
+
+impl AllocatedSize for u64 {
+    fn allocated_size(&self) -> usize {
+        8
+    }
+}
+
+impl AllocatedSize for String {
+    fn allocated_size(&self) -> usize {
+        self.capacity()
+    }
+}
+
+impl AllocatedSize for AppIdentifier {
+    fn allocated_size(&self) -> usize {
+        match self {
+            AppIdentifier::AppId(app_id, None) => {
+                std::mem::size_of::<AppIdentifier>() + app_id.as_ref().to_string().capacity()
```

Comment:
Note that you are cloning a string to note down how much capacity it needs in the heap. This isn't realistically going to be done at scale, so you'd have to estimate the capacity in the heap based off the string's length.